wdl-engine/src/eval/v1/task.rs

```diff
@@ -0,0 +1,929 @@
+
+                            command = self.evaluate_command(
+                                document,
+                                execution.as_mut(),
+                                &scopes,
+                                task,
+                                section,
+                                &mapped_paths,
+                            )?;
+                        }
+                        TaskGraphNode::Runtime(section) => {
+                            assert!(
+                                requirements.is_none(),
+                                "requirements should not have been evaluated"
+                            );
+                            assert!(hints.is_none(), "hints should not have been evaluated");
+
+                            let (r, h) = self.evaluate_runtime_section(
+                                document,
+                                execution.as_ref(),
+                                &scopes,
+                                task,
+                                section,
+                                inputs,
+                            )?;
+
+                            requirements = Some(r);
+                            hints = Some(h);
+                        }
+                        TaskGraphNode::Requirements(section) => {
+                            assert!(
+                                requirements.is_none(),
+                                "requirements should not have been evaluated"
+                            );
+                            requirements = Some(self.evaluate_requirements_section(
+                                document,
+                                execution.as_ref(),
+                                &scopes,
+                                task,
+                                section,
+                                inputs,
+                            )?);
+                        }
+                        TaskGraphNode::Hints(section) => {
+                            assert!(hints.is_none(), "hints should not have been evaluated");
+                            hints = Some(self.evaluate_hints_section(
+                                document,
+                                execution.as_ref(),
+                                &scopes,
+                                task,
+                                section,
+                                inputs,
+                            )?);
+                        }
+                    }
+
+                    current += 1;
+                }
+
+                let requirements = requirements.unwrap_or_default();
+                let hints = hints.unwrap_or_default();
+
+                // TODO: check call cache for a hit. if so, skip task execution and use cache
+                // paths for output evaluation
+
+                let status_code = execution.spawn(command, &requirements, &hints)?.await?;
+
+                // TODO: support retrying the task if it fails
```

Comment:
Ah I was wondering about this. I thought maybe I missed something. Glad to see I've been following along correctly 😅